CI: `post-kubernetes-push-e2e-agnhost-test-images` is failing (`gcr.io/k8s-staging-e2e-test-images/agnhost:2.46-linux-amd64 is a manifest list`)

### Which jobs are failing?

agnhost was updated to v2.46 in:
- https://github.com/kubernetes/kubernetes/pull/123258

However, `post-kubernetes-push-e2e-agnhost-test-images` is failing: https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/post-kubernetes-push-e2e-agnhost-test-images/1757425555977801728
```
...
#27 exporting to image
#27 exporting layers
#27 exporting layers 2.9s done
#27 exporting manifest sha256:db8e9d1930d4799d474ea9228d4957d54053c107e10385d77021ef44f567fc69 done
#27 exporting config sha256:037153f61da4211d6ec5f00f694639ce15a0bbebf12fa5c86eea059fc3c0b07c done
#27 exporting attestation manifest sha256:744b616c276ff7f6518476765c2457e0b2664225ef798ce7e93874450807310b
#27 exporting attestation manifest sha256:744b616c276ff7f6518476765c2457e0b2664225ef798ce7e93874450807310b 0.0s done
#27 exporting manifest list sha256:29bf785c1a8338d5f7a8be8cae8a2435b94370eecbaa586a61e08454095940cb done
#27 pushing layers
#27 pushing layers 5.9s done
#27 pushing manifest for gcr.io/k8s-staging-e2e-test-images/agnhost:2.46-windows-amd64-ltsc2022@sha256:29bf785c1a8338d5f7a8be8cae8a2435b94370eecbaa586a61e08454095940cb
#27 pushing manifest for gcr.io/k8s-staging-e2e-test-images/agnhost:2.46-windows-amd64-ltsc2022@sha256:29bf785c1a8338d5f7a8be8cae8a2435b94370eecbaa586a61e08454095940cb 4.3s done
#27 DONE 13.2s
/workspace/test/images
/workspace/test/images/agnhost /workspace/test/images
gcr.io/k8s-staging-e2e-test-images/agnhost:2.46-linux-amd64 is a manifest list
make: *** [Makefile:43: all-build-and-push] Error 1
ERROR
ERROR: build step 0 "gcr.io/k8s-staging-test-infra/gcb-docker-gcloud:v20230623-56e06d7c18" failed: step exited with non-zero status: 2
--------------------------------------------------------------------------------

2024/02/13 15:34:04 Failed to run some build jobs: [error running [gcloud builds submit --verbosity info --config /home/prow/go/src/github.com/kubernetes/kubernetes/test/images/cloudbuild.yaml --substitutions _PULL_BASE_REF=master,_PULL_BASE_SHA=c3b2295c3ae5aafc884b37eed5f4c0c4109f26bd,_WHAT=agnhost,_GIT_TAG=v20240213-v1.30.0-alpha.1-173-gc3b2295c3a --project k8s-staging-e2e-test-images --gcs-log-dir gs://k8s-staging-e2e-test-images-gcb/logs --gcs-source-staging-dir gs://k8s-staging-e2e-test-images-gcb/source gs://k8s-staging-e2e-test-images-gcb/source/960f9db9-7566-469c-a5a0-9eb2cf4549e3.tgz]: exit status 1]
```


### Which tests are failing?

N/A

### Since when has it been failing?

2024-02-14


### Testgrid link

https://testgrid.k8s.io/sig-k8s-infra-gcb#post-kubernetes-push-e2e-agnhost-test-images

### Reason for failure (if possible)

_No response_

### Anything else we need to know?

_No response_

### Relevant SIG(s)

/sig testing

--- a/test/images/image-util.sh
+++ b/test/images/image-util.sh
@@ -185,7 +185,8 @@ build() {
         "${SED}" -i '/CROSS_BUILD_/d' Dockerfile
     fi
 
-    docker buildx build --progress=plain --no-cache --pull --output=type="${output_type}" --platform "${os_name}/${arch}" \
+    # `--provenance=false --sbom=false` is set to avoid creating a manifest list: https://github.com/kubernetes/kubernetes/issues/123266
+    docker buildx build --progress=plain --no-cache --pull --output=type="${output_type}" --platform "${os_name}/${arch}" --provenance=false --sbom=false \
         --build-arg BASEIMAGE="${base_image}" --build-arg REGISTRY="${REGISTRY}" --build-arg OS_VERSION="${os_version}" \
         -t "${REGISTRY}/${image}:${TAG}-${suffix}" -f "${dockerfile_name}" \
 	--label "image_version=${TAG}" --label "commit_id=${GIT_COMMIT_ID}" \

